File: javav2/example_code/bedrock-runtime/src/main/java/com/example/bedrockruntime/libs/ImageTools.java
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.bedrockruntime.libs;

import javax.imageio.ImageIO;
import javax.swing.ImageIcon;
import javax.swing.JFrame;
import javax.swing.JLabel;
import java.awt.image.BufferedImage;
import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.util.Base64;

public class ImageTools {

    public static void displayImage(String base64ImageData) {
        byte[] imageData = Base64.getDecoder().decode(base64ImageData);
        displayImage(imageData);
    }

    public static void displayImage(byte[] imageData) {
        try {
            BufferedImage image = ImageIO.read(new ByteArrayInputStream(imageData));
            JFrame frame = new JFrame("Image");
            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);

            JLabel label = new JLabel(new ImageIcon(image));
            frame.getContentPane().add(label);

            frame.pack();
            frame.setVisible(true);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }
    }
}

File: kotlin/services/bedrock-runtime/src/main/kotlin/com/example/bedrockruntime/libs/ImageTools.kt
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package com.example.bedrockruntime.libs

import java.io.ByteArrayInputStream
import java.io.IOException
import javax.imageio.ImageIO
import javax.swing.ImageIcon
import javax.swing.JFrame
import javax.swing.JLabel

/**
 * Utility object for handling image-related operations.
 */
object ImageTools {
    /**
     * Displays a byte array as an image in a new window.
     *
     * Creates a new JFrame window that displays the image represented by the provided byte array.
     * The window will close the application when closed (EXIT_ON_CLOSE).
     *
     * @param imageData The image data as a byte array
     * @throws RuntimeException if there is an error reading the image data
     */
    fun displayImage(imageData: ByteArray) {
        try {
            val image = ImageIO.read(ByteArrayInputStream(imageData))
            JFrame("Image").apply {
                defaultCloseOperation = JFrame.EXIT_ON_CLOSE
                contentPane.add(JLabel(ImageIcon(image)))
                pack()
                isVisible = true
            }
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}
